Ch/new moment

GlowingEgg opened the pull request:
Lays out the Make a Moment Activity and sets it up for all future interaction.

GlowingEgg committed:
Completes the VideoView displayed when a Moment is selected.
Also begins using the legitimate Moments in Time Vimeo account.

GlowingEgg committed:
Sets up the MakeAMoment Activity.

GlowingEgg committed:
Adds the menu to the MakeAMomentActivity.

GlowingEgg committed:
Sets up notes.
Notes now exist in the Moment and on the MakeAMomentActivity screen.

GlowingEgg committed:
Changes the MakeAMomentActivity to use a single RecyclerView for all views on screen.
Stops using the CoordinatorLayout in activity_make_a_moment, starts using a single RecyclerView, and changes the adapter to reflect that. Adds the corresponding holders.

0xc00010ff commented on app/src/main/java/com/tikkunolam/momentsintime/MakeAMomentAdapter.java:
```diff
@@ -0,0 +1,419 @@
+    final int NOTES_TEXT = 6;
+    final int NOTES_PROMPT = 7;
+    final int NOTE = 8;
+
```
This is functionally the same, but most people will expect that you use an [enum](http://stackoverflow.com/a/4709224/1922780) here, since this is a *set of potential values*
```
public enum CreateMomentSection {
    InterviewingTitle(0),
    InterviewingEdit(1),
    DescriptionTitle(2),
    DescriptionEdit(3);
    ...and so on

    private final int adapterPosition;

    CreateMomentSection(int position) {
        this.adapterPosition = position;
    }

    public int position() {
        return this.adapterPosition;
    }
}
```

0xc00010ff commented on app/src/main/java/com/tikkunolam/momentsintime/MakeAMomentAdapter.java:
```diff
@@ -0,0 +1,419 @@
+    Context mContext;
+
+    // the View filling the RecyclerView cells
+    int mResource;
```
What is this for?

0xc00010ff commented on app/src/main/java/com/tikkunolam/momentsintime/MakeAMomentAdapter.java:
```diff
@@ -0,0 +1,419 @@
+
+        switch(position) {
+
+            case 0:
```
See enums comment above. You'll want those here.

0xc00010ff commented on app/src/main/java/com/tikkunolam/momentsintime/MakeAMomentAdapter.java:
```diff
@@ -0,0 +1,419 @@
+
+        for(int i = 0; i < 8; i++) {
+
+            mIdentifierList.add(i, "not a note");
```
There's something very smelly about this

0xc00010ff reviewed (changes requested):
See comments on enums.

Can you add like 5 dummy notes to show note functionality in the list? The app will start with 5 or so default notes anyway.

GlowingEgg committed:
Changes the MakeAMomentActivity to move the logic that orders the views out of the Adapter.

GlowingEgg committed:
Sets up the MakeAMomentActivity to deal with user interaction.
Modifies the SectionPromptHolder to determine what callback to attach to its textView's onClick, based on the text provided. Implements infinite scroll on the Moment lists.

0xc00010ff reviewed (approved):
![](https://media0.giphy.com/media/k39w535jFPYrK/200.gif#16-grid1)

Notes: 
Flirt with enums. 
"*private* void function() { ... }"
